src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -181,17 +181,39 @@ class ContentExplorer extends Component {
     }
 
     componentDidUpdate({ initialFoldersPath: prevInitialFoldersPath }) {
-        const { initialFoldersPath } = this.props;
+        const { initialFoldersPath, isSelectAllAllowed } = this.props;
+        const { isSelectAllChecked } = this.state;
 
         if (prevInitialFoldersPath !== initialFoldersPath) {
             this.handleFoldersPathUpdated(initialFoldersPath);
         }
+        if (!this.isLoadingItems() && isSelectAllAllowed) {
+            const areAllCurrentItemsSelected = this.areAllCurrentItemsSelected();
+            if (areAllCurrentItemsSelected !== isSelectAllChecked) {
+                this.setState({ isSelectAllChecked: areAllCurrentItemsSelected });
+            }
+        }
     }
 
     componentWillUnmount() {
         document.removeEventListener('click', this.handleDocumentClick, true);
     }
 
+    areAllCurrentItemsSelected = () => {
+        const { items } = this.props;
+        const { selectedItems } = this.state;
+
+        const someItemIsNotSelected = items.some(item => {
+            return !selectedItems[item.id];
+        });
```

Comment:
Ah yes using .every is a better implementation. I originally used .every then changed to .some because I mistakenly thought that their time complexity would be different. I will change to use .every